trio/_dtls.py

```diff
@@ -697,7 +739,9 @@ async def handle_client_hello_untrusted(endpoint, address, packet):
         endpoint._incoming_connections_q.s.send_nowait(stream)
 
 
-async def dtls_receive_loop(endpoint_ref, sock):
+async def dtls_receive_loop(
+    endpoint_ref: weakref.ReferenceType[DTLSEndpoint], sock: _SocketType
```

Comment:
Huh, `weakref.ReferenceType`, I remember using `weakref.ref` as a generic. I looked at it and supposedly it's just a convenience for `ReferenceType` but it does look nicer!

Reply:
which one looks nicer? ig `ReferenceType` is a bit verbose unless you do a `from x import y` import

Reply:
Sorry, IMO weakref.ref looks nicer fully-qualified. But tbf this is a nit. Up to you.